flake: sporadic test failures from TestMachine

`machine1` testsuite sporadically fails with:

```
=== RUN   TestMachine
--- FAIL: TestMachine (0.03s)
	dbus_test.go:102: unexpectedly found machine named machined-test-mykjbyxx
```

I think this could be because `TerminateMachine` is asynchronous and we are racing against termination, but I didn't check server-side code in systemd to confirm that. If that is the case, testcode should be augmented with retry+timeout logic.

--- a/machine1/dbus_test.go
+++ b/machine1/dbus_test.go
@@ -97,7 +97,13 @@ func TestMachine(t *testing.T) {
 		t.Fatal(tErr)
 	}
 
-	machine, getErr = conn.GetMachine(machineName)
+	for i := 1; i <= 10; i++ {
+		machine, getErr = conn.GetMachine(machineName)
+		if len(machine) == 0 && getErr != nil {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
 	if len(machine) != 0 {
 		t.Fatalf("unexpectedly found machine named %s", machineName)
 	} else if getErr == nil {

